File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const rep = (str)=>{
        return str.replace(/2|1|0/g,v=> +v>0 ? '#' : ' ');
    }
    
   let a1 = arr1.map((v,i)=> +v.toString(2) + +arr2[i].toString(2) +'')
   let answer = a1.map(v=> rep('0'.repeat(n-v.length)+v))
    
    return answer
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        for (int i = 0; i < arr1.length; i++) {

            String str1 = Integer.toString(arr1[i], 2);
            String str2 = Integer.toString(arr2[i], 2);

            str1 = "0".repeat(n - str1.length()) + str1;
            str2 = "0".repeat(n - str2.length()) + str2;

            StringBuilder sb = new StringBuilder();

            for (int j = 0; j < n; j++) {
                if (str1.charAt(j) == '0' && str2.charAt(j) == '0') {
                    sb.append(" ");
                } else {
                    sb.append("#");
                }
            }
            answer[i] = sb.toString();
        }
        return answer;
    }
}